source/develop/release-management/features/network/copy-host-networking.html.md

```diff
@@ -0,0 +1,125 @@
+---
+title: Copy Host Networking
+category: feature
+authors: Ales Musil
+feature_name: Copy Host Networking
+feature_modules: copy-host-networking
+feature_status: In Development
+---
+
+# Copy Host Networking
+
+## Summary
+
+Ordinarily, adding hosts in oVirt in the UI is tedious because you, as an administrator, have to set up the host 
+network on each new host.
+
+The Copy Host Networking feature makes adding new hosts easier because you can configure the network on a single host
+and then copy that configuration to any new host that has the same layout of network interfaces.
+
+You can use this feature to perform initial deployments and to update running environments. It is available as a button
+in the Administration Portal UI, and as a REST API method.
+
+
+### Owner
+
+- Feature Owner: Ales Musil (amusil)
+- E-mail: <amusil@redhat.com>
+
+### Example flow
+
+After creating a cluster, you add several hosts (`Host0, Host1, Host2`). 
+You configure the network settings for `Host0`. The two other hosts have the same network 
+interface layout as `Host0`, so you copy the network configuration from `Host0` to `Host1` and `Host2`.
+
+In the Admin Portal, you go to the `Host1` and `Host2` network interfaces and click
+the `Copy Host Networking` button. In the popup window that appears,
+you select the host, `Host0`, from which to copy the configuration.
+
+Alternatively, with the REST API, you can use the `copynetworks` method for `Host1` and `Host2`. 
+This method takes a single parameter, the host ID from which to copy the configuration (`Host0`). 
+
+### Limitations
+
+* You can copy host configurations only within the same cluster. 
+
+* You cannot copy network configurations that contain static IP addresses: 
+This sets the boot protocol in the resulting network to `none`.
+
+* Copying a configuration to a host with the same interface names but different physical network connections will
+produce a bad configuration. Only copy configurations to hosts that have the same interface names and physical 
+network connections. 
+
+* The destination host has to have an equal or greater number of interfaces as the original host. Otherwise,
+the feature cannot determine what to do.
+
+* The `ovirtmgmt` network stays on the original interface which is excluded from the operation. 
+
+
+### Implementation
+
+A new command `CopyNetworks` is added to the engine. That has destination and source host as parameters.
```

Comment:
The engine has a new method, `CopyNetworks`, that copies the network configuration of the source host (`SourceHost`) and uses it to overwrite the network configuration of the destination host (`DestinationHost`).

This method takes two parameters, the `SourceHost` ID and the `DestinationHost` ID. 
 
Given the parameters, the method validates that the following items are correct:
  * Both hosts are in the same cluster. 
  * The destination host has an equal or higher number of interfaces than the source host. 


(**NOTE:** These interface names and physical connections to the datacenter must be identical. Otherwise, this sorting process produces a bad network configuration on the destination host.)

The method prepares a set of network attachment definitions for the `DestinationHost` by doing the following:
 
* Lists the interfaces on the `SourceHost` that don't have bonds.
* Lists all of the interfaces on the  `DestinationHost`. 
* Sorts both lists by name.
* Excludes interface/bonds connected to the `ovirtmgmt` network.
 * Pairs all of the bond slaves from the `SourceHost` with their corresponding partners on the `DestinationHost`. 
*  Creates bond definitions in a configuration for the `DestinationHost`.	 
* Retrieves a list of all interfaces with bonds from `SourceHost`.
* Excludes bond slaves from both lists.
* Shortens the `DestinationHost` list to match the same length as `SourceHost` list. 	 
* Copies the network attachments for all interfaces in `SourceHost` list.
* Walks through the new network attachments list and updates any static IP address configurations.
* Sends `HostSetupNetworks` with new attachments and bonds for `DestinationHost` to complete everything in a single transaction. 
* If there is an error, it reverts the configuration changes. 	 
 
  
The Admin Portal has a new button called `Copy Host Networking`. Clicking the button opens a dialog window. There, you can select the source host from which to copy the configuration.

Reply:
Done, thanks